plugin.video.dazn/resources/lib/simple_requests/api.py

```diff
@@ -0,0 +1,195 @@
+             stream=None,
+             verify=None,
+             cert=None,
+             json=None):
+    if not headers:
+        headers = {}
+        pass
+
+    url = urllib.quote(url, safe="%/:=&?~#+!$,;'@()*[]")
+
+    handlers = []
+
+    import sys
+    # starting with python 2.7.9 urllib verifies every https request
+    if False == verify and sys.version_info >= (2, 7, 9):
+        import ssl
+
+        ssl_context = ssl.create_default_context()
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+        handlers.append(urllib2.HTTPSHandler(context=ssl_context))
+        pass
+
+    # handlers.append(urllib2.HTTPCookieProcessor())
+    # handlers.append(ErrorHandler)
+    if not allow_redirects:
+        handlers.append(NoRedirectHandler)
+        pass
+    opener = urllib2.build_opener(*handlers)
+
+    query = ''
+    if params:
+        for key in params:
+            value = params[key]
+            if isinstance(value, str):
+                value = value.decode('utf-8')
+                pass
+            params[key] = value.encode('utf-8')
+            pass
+        query = urllib.urlencode(params)
+        pass
+    if query:
+        url += '?%s' % query
+        pass
+    request = urllib2.Request(url)
+    if headers:
+        for key in headers:
+            request.add_header(key, str(unicode(headers[key]).encode('utf-8')))
+            pass
+        pass
+    if data or json:
+        if headers.get('Content-Type', '').startswith('application/x-www-form-urlencoded') and data:
+            # transform a string into a map of values
+            if isinstance(data, basestring):
+                _data = data.split('&')
+                data = {}
+                for item in _data:
+                    name, value = item.split('=')
+                    data[name] = value
+                    pass
+                pass
+
+            # encode each value
+            for key in data:
+                data[key] = data[key].encode('utf-8')
+                pass
+
+            # urlencode
+            request.data = urllib.urlencode(data)
+            pass
+        elif headers.get('Content-Type', '').startswith('application/json') and data:
+            request.data = real_json.dumps(data).encode('utf-8')
+            pass
+        elif json:
+            request.data = real_json.dumps(json).encode('utf-8')
+            pass
+        else:
+            if not isinstance(data, basestring):
+                data = str(data)
+                pass
+
+            if isinstance(data, str):
+                data = data.encode('utf-8')
+                pass
+            request.data = data
+            pass
+        pass
+    elif method.upper() in ['POST', 'PUT']:
+        request.data = "null"
+        pass
+    request.get_method = lambda: method
+    result = Response()
+    response = None
+    try:
+        response = opener.open(request)
+    except urllib2.HTTPError, e:
```

Comment:
preferred syntax is `except urllib2.HTTPError as e` which is also python3 compatible